Folding paper Formula is t(n) = t(0) * 2^(n): Since we know the variables t(n) - thickness, and t(0) - 0.00008m ; we need to find the power (n).
I will use logarithm which is a math function in python to find (n).
 1. log(t(n)) = log(t(0)) + log(2^(n))
 2. log(t(n)) = log(t(0)) + n*log(2)
 3. n*log(2) = log(t(n)) - log(t(0))
 4. n = (log(t(n)) - log(t(0))) / log(2)- formula to find n- (no of folds)

# Problem 1:
t(n) = 3776 m
t(0) = 0.00008 m

In [38]:
import math # library to use logarithm
n = (math.log(3776) - math.log(0.00008)) / math.log(2) # log to calculate (n)according to the formula
no_of_folds = math.ceil(n) # function to round up n(if it contains decimals) to the next whole number
print(no_of_folds) 

26


To exceed mount fuji the paper will be folded 26 times.

# Problem 2:
Function that accepts 2 arguments - the height input(t(n)) and paper thicknes before fold(t(0)) and outputs the minimum number of paper folds required to exceed it.

In [46]:
def n_folds_to_reach_height(height, thickness):
    n = (math.log(height) - math.log(thickness)) / math.log(2) # calculate no of folds given height and thickness
    no_of_folds = math.ceil(n) # function to round up n(if it contains decimals) to the next whole number
    print("The minimum number of folds required to exceed height is:",no_of_folds)
    
#height to nearest star is 4.0175*10^16m, thickness t(0) = 0.00008m


n_folds_to_reach_height((4.0175*(pow(10, 16))), 0.00008)

The minimum number of folds required to exceed height is: 69


It will take 69 folds to reach proxima centauri.

# Problem 3:
Function that calculates the length of the paper required to fold the paper and reach any thickness.
The function takes height(meters) and thickness of paper before fold(meters), and outputs the length of paper required in meters.

In [22]:
#1. Calculate the length of paper required to fold the paper to the thickness equal to moon's distance
import math
def paper_length_required(height, thickness):# function to get length takes height and thickness as arguments
    print("The length is:")
    def n_folds_to_reach_height():# inner function that calculates number of folds
        n = (math.log(height) - math.log(thickness)) / math.log(2)
        no_of_folds = math.ceil(n)
        length = ((math.pi*thickness)/6)*((pow(2,no_of_folds)) + 4 )*((pow(2, no_of_folds)) - 1)# formula to calculate length of paper required
        print(length,":meters")
    return n_folds_to_reach_height()# returns the output of inner function to the outer function
# distance to moon = 384400000 meters
# paper thickness = 0.00008 meters
paper_length_required(384400000, 0.00008 )

The length is:
3.240919444170781e+21 :meters


The length of paper required to fold thickness equal to moons distance (3.24 * 10 ^21) meters

In [21]:
#2. Calculate the length of paper required to fold the paper to the thickness equal to height of mount fuji
import math
def paper_length_required(height, thickness):# function to get length takes height and thickness as arguments
    print("The length is:")
    def n_folds_to_reach_height():# inner function that calculates number of folds
        n = (math.log(height) - math.log(thickness)) / math.log(2)
        no_of_folds = math.ceil(n)
        length = ((math.pi*thickness)/6)*((pow(2,no_of_folds)) + 4 )*((pow(2, no_of_folds)) - 1)# formula to calculate length of paper required
        print(length,":meters")
    return n_folds_to_reach_height()# returns the output of inner function to the outer function
# distance to mt fuji = 3776 meters
# paper thickness = 0.00008 meters
paper_length_required(3776, 0.00008 )

The length is:
188646348487.24017 :meters


The length of paper required to fold thickness equal to height of mount fuji 188646348487.24017 meters

In [23]:
#3. Calculate the length of paper required to fold the paper to the thickness equal to distance to proxima centauri
import math
def paper_length_required(height, thickness):# function to get length takes height and thickness as arguments
    print("The length is:")
    def n_folds_to_reach_height():# inner function that calculates number of folds
        n = (math.log(height) - math.log(thickness)) / math.log(2)
        no_of_folds = math.ceil(n)
        length = ((math.pi*thickness)/6)*((pow(2,no_of_folds)) + 4 )*((pow(2, no_of_folds)) - 1)# formula to calculate length of paper required
        print(length,":meters")
    return n_folds_to_reach_height()# returns the output of inner function to the outer function
# distance to proxima centauri = 4.0175*10^16 meters
# paper thickness = 0.00008 meters
paper_length_required((4.0175*(pow(10,16))), 0.00008 )

The length is:
1.4595803601100348e+37 :meters


The length of paper required to fold thickness equal to distance of nearest star proxima centauri is (1.46 * 10^37) meters